### PR TITLE
Syndicate Fedora

### PR DESCRIPTION
**Pull request by OliBomby**
Adds syndicate fedoras to the traitor uplink.
Can be found under useless badassery.

Able to extend a sharp edge to make it a weapon.
Intended for throwing and leaving it behind because it's embedded.

It's my first code :)
All sprites included.

### CODE REVIEW

**Comment by Alek2ander** [comment]
6 is a huge cost for such a gimmicky item. You throw it once, it embeds, now you don't have it anymore. And the person it hits can then remove it and use it against you.

**Comment by Nirnael** [comment]
Sprite PRs need a screenshot.

**Comment by B1gb3ast** [comment]
@Alek2ander I think 6 telecrystals is fine for validsalad. You complete your objectives, have some left over and then just order a daft fedora to make you look cool. The fedora throw is just like, an ace up your sleeve.

alright. you got the traitor in maint, his gun is out of ammo, you rush at him with a baton - he throws his fedora into your face and runs the fuck away, making you bleed and unable to really chase him unless you wanna risk dying.

You could Always lower the TC, though. an energy sword costs 6 too.

**Comment by OliBomby** [comment]
http://puu.sh/qcQzB/151460968e.png
http://puu.sh/qcQB9/f3d23a43be.png
http://puu.sh/qcQXO/f8b5847721.png

Do know that the fedora throw can do huge damage. Able to crit from just the hit and embed damage on some occasions.

**Comment by X-TheDark** [comment]
Just in case anyone is interested in the number side of things
- This item would do 40 damage on hit (armor reductions can apply) + 16 damage due to embedding (weigh class \* embedded_impact_pain_multiplier (it's 4)).

After which, every tick:
- There's a 15% chance for 16 damage (weight class \* embedded_pain_multiplier).
- 5% chance of it falling off, doing 20 damage (weight class \* embedded_fall_pain_multiplier (which is 5)).

Hilariously enough, both of the on tick effects can happen at the same time - you get the on-tick damage, after which the object falls off, doing the extra falling off damage.

**Comment by Notamaniac** [comment]
For the same price, you can buy a syndi minibomb, which'll will probably outright kill your target, if not, stunning them near a breach (unless you decided to throw it uncooked and let people run away).

Have you also considered what'd happen if you miss? Or if the person sees the hat black hat and decides to try catching it? For 6TC, there are far better less memey options that probably won't backfire on you horribly.

The only possible way I can see this having value is if you can recall it into your hands like the ninja's katana.

Sprites also need more shading, the sharp edge just looks like a grey trim, add some lighter shades of grey to give it some shine. I might also consider extending the edge to two pixels to stop it looking like a trim and more of a blade, but I have no idea how'd that work out (edit : spikes might work too). The hat also needs the patented SS13 'slightly darker outline that isn't black', and a little more shading around the dimples, it's way too harsh a contrast. Said dimples are also triggering my NEVER USE PURE BLACK mantra.

**Comment by ParagonPrime** [comment]
Recall on a timer sounds good.

Also, blade-extended sprites should be a little more... blade-like.

Like, long gray wedges sticking out of the sides of the hat.

**Comment by FluffySurvivor** [comment]
why does everyone overprice their new items? 
This is 2-3 TC MAX.

**Comment by OliBomby** [comment]
As sprite and concept I was really thinking this guy:
http://puu.sh/qcVha/be08685a9a.jpg

Guess the TC cost has to be lower. Is 3 TC OK?

**Comment by B1gb3ast** [comment]
Why is everybody getting their jimmies rustled over an item in pointless badassery category though.

Olibomby isn't a spriter either, and I'm pretty sure the syndicate fedora is litteraly a recolored detective's fedora.

**Comment by Alek2ander** [comment]
What if instead of embedding it bounces back to you like a boomerang? Who'd want to lose a classy fedora?

**Comment by FluffySurvivor** [comment]
> pointless badassery category though.

Because it turns into a noob bait

**Comment by TheSadFox** [comment]
40 throwforce 15 force 100% embed chance item is hillariously deadly. But we already have items that fulfill the role of an antag throwing weapon (throwing stars), don't really see any reason to add more. 

> What if instead of embedding it bounces back to you like a boomerang?

This + have it a high chance to dismember limbs. I'd totally buy that shit.

**Comment by B1gb3ast** [comment]
a fedora that is a boomerang.

that makes no sense.

wat.

But yeah, we need a boomerang code. boomerang stunbatons, boomerang fedora blades, boomerang eswords. Somebody make a request already.

**Comment by TheSadFox** [comment]
> that makes no sense

Welcome to ss13 man.

**Comment by OliBomby** [comment]
I think boomerang dismember is pretty cool.
But I don't have the skills to code that.
Let's just get this in first and you guys may rework it. :)

**Comment by TheSadFox** [comment]
Boomerang code you could probably look up in #329.

**Comment by OliBomby** [comment]
I'll look into it later.

**Comment by ghost** [comment]
This hat is deadlier than the throwing box of ninja stars, in all fairness. Catching an object is hard and doesn't work with embed objects like ninjastars, bola, and now this hat.

It is also under pointless badassery, and thus supposed to be useless, like the balloon or throwing cards. Noob bait or not you don't need to pat antags on the dick to make them feel better.

File: code/modules/uplink/uplink_item.dm [hunk 1]
```diff
@@ -1104,6 +1104,12 @@ var/list/uplink_items = list() // Global list so we only initialize this once.
 	item = /obj/item/weapon/storage/fancy/cigarettes/cigpack_syndicate
 	cost = 1
 
+/datum/uplink_item/badass/syndiefedora
+	name = "Syndicate Fedora"
+	desc = "This Syndicate Fedora of micro-woven adamantium silk is sure the prove your style!"
```
Comment (ShadowDeath6) [inline code/modules/uplink/uplink_item.dm]: might want to make that "to prove" rather than "the prove"

File: code/modules/clothing/head/jobs.dm [hunk 2]
```diff
@@ -79,6 +79,38 @@
...
+/obj/item/clothing/head/det_hat/evil
+	name = "syndicate fedora"
+	icon_state = "syndicate_fedora"
+	desc = "A suspicious black fedora with a red band."
+	embed_chance = 100
+	sharpness = IS_SHARP
+	w_class = 4
+	hitsound = 'sound/weapons/Genhit.ogg'
+	attack_verb = list("poked")
+	var/extended
```
Comment (Cruix) [inline code/modules/clothing/head/jobs.dm]: extended should be explicitly set to 0, sharpness and attack_verb should be consistent with what they are set to in attack_self when the blade is retracted.

File: code/modules/clothing/head/jobs.dm [hunk 3]
```diff
@@ -79,6 +79,38 @@
...
+
+/obj/item/clothing/head/det_hat/evil/attack_self(mob/user)
+	extended = !extended
+	playsound(src.loc, 'sound/weapons/batonextend.ogg', 50, 1)
+	if(extended)
+		force = 15
+		embedded_pain_multiplier = 4
+		throw_speed = 4
+		throwforce = 40
+		embed_chance = 100
```
Comment (Cruix) [inline code/modules/clothing/head/jobs.dm:100]: If you are going to set this to 100  here, you may as well set it to 0 in the else and in the initial values.

File: code/modules/clothing/head/jobs.dm [hunk 4]
```diff
@@ -79,6 +79,39 @@
 		else
 			M << "You just took a candy corn! You should wait a couple minutes, lest you burn through your stash."
 
+/obj/item/clothing/head/det_hat/evil
+	name = "syndicate fedora"
+	icon_state = "syndicate_fedora"
+	desc = "A suspicious black fedora with a red band."
+	w_class = 4
+	throw_speed = 4
+	embed_chance = 0
```
Comment (Super3222) [inline code/modules/clothing/head/jobs.dm]: Everything starts with an embed_chance of 0. You don't need to define it here!

**Comment by OliBomby** [comment]
I manually fixed the dmi files so it can merge again. Please merge before something else breaks it again.

**Comment by Nirnael** [comment]
This can be metagamed so hard if the fedora has a snowflake icon. Better just add a blade icon to the existing one. Or change the existing one.

**Comment by Super3222** [comment]
> This can be metagamed so hard if the fedora has a snowflake icon.

It's pointless badassery though, that's the entire name of the game. I'm not sure about keeping the embedding part of it because you know, pointless badassery.

**Comment by Nirnael** [comment]
It's pointless badassery that can still be metagamed and serves as a weapon.

**Comment by ghost** [comment]
Everything in the pointless badassery section of the uplink is obvious. Cards, smokes, GIANT ASS BALLOON

**Comment by Nirnael** [comment]
I thought the balloon was metashielded?

**Comment by ParagonPrime** [comment]
The balloon IS metashielded.

I mean, whatever sprite it takes, this should also be metashielded (until it sprouts blades). Because, you know, it's a hat.

Kinda like how noslips (even when their function is discovered), the syndicate duffel/smuggler satchel, and etc. are also metashielded. Because they aren't actually illegal by themselves.

**Pull request merged**